### PR TITLE
Reduce allocations / clean up code in Process (+ fix name clash with Formatter)

### PR DESCRIPTION
**Pull request by mathias-lang-sociomantic**
Importing both `Formatter` and `Process` led to a needless clash. This fixes it along with a few other code oddities.

### CODE REVIEW

File: src/ocean/sys/Process.d [hunk 1]
```diff
@@ -866,7 +856,7 @@ class Process
 
             // validate the redirection flags
             if((_redirect & (Redirect.OutputToError | Redirect.ErrorToOutput)) == (Redirect.OutputToError | Redirect.ErrorToOutput))
-                throw new ProcessCreateException(_args[0], "Illegal redirection flags", __FILE__, __LINE__);
+                throw new ProcessCreateException(_args[0], "Illegal redirection flags");
```
Comment (leandro-lucarella-sociomantic) [inline src/ocean/sys/Process.d:870]: Is this really related to the use of `Formatter`?

File: src/ocean/sys/Process.d [hunk 2]
```diff
@@ -1596,14 +1586,12 @@ class Process
  */
 class ProcessCreateException: ProcessException
 {
-    public this(cstring command, istring file, uint line)
+    public this (cstring command, istring message = SysError.lastMsg,
+                 istring file = __FILE__, uint line = __LINE__)
```
Comment (leandro-lucarella-sociomantic) [inline src/ocean/sys/Process.d]: I guess not, can you split this change in a separate commit?

**Comment by mathias-lang-sociomantic** [comment]
Splitted changes to be more atomic

**Review by mihails-strasuns-sociomantic — APPROVED** [review]
Thanks, very nice!
